lib/watir/elements/element.rb

```diff
@@ -495,6 +479,48 @@ def browser
 
     protected
 
+    def wait_for_exists(timeout = nil)
+      return assert_exists unless Watir.relaxed_locate?
+
+      timeout ||= Watir.default_timeout
+      return if exists? # Performance shortcut
+
+      assert_element_found unless Watir.always_locate? # don't re-lookup element
+      timeout_off = timeout == 0
+
+      begin
+        @parent.wait_for_exists(timeout)
```

Comment:
Does this mean that chained element calls can wait longer than the specified timeout? For example, `browser.div.link.click` can:
* Wait up to 29 seconds for the `div` and then (assuming it appears)
* Wait up to another 30 seconds for the `link`?

I would expect the timeout to be applied to the entire action, which to a user is just clicking the link.